### PR TITLE
Collection ownername

### LINKED ISSUE #1159: Add a collection overview to package collections
It currently says "This collection does not have an overview" when adding a collection to Xcode.

Screenshot 2021-06-08 at 12 59 46@2x

It should say "A collection of packages authored by Vapor from the Swift Package Index"

**Note:** Use the full owner name rather than just the owner username when building this description

### LINKED ISSUE #1160: Use full owner name when building package collections
We should use the full owner name here rather than the username.

Screenshot 2021-06-08 at 13 03 18@2x

### PR DESCRIPTION
**Pull request by finestructure**
Supersedes #1192 

Fixes #1159
Fixes #1160

### CODE REVIEW

File: Sources/App/Core/PackageCollection+generate.swift [hunk 1]
```diff
@@ -55,21 +55,48 @@ extension PackageCollection {
...
-                    packages: $0,
+                    packages: packages,
                     formatVersion: .v1_0,
                     revision: revision,
                     generatedAt: Current.date(),
                     generatedBy: authorName.map(Author.init(name:)))
             }
     }
+
+    static func ownerName(packages: [App.Package]) -> String {
```
Comment (at-syot) [inline Sources/App/Core/PackageCollection+generate.swift:81]: look better.

**Review by at-syot — APPROVED** [review]

**Comment by finestructure** [comment]
Draft until I've added an explicit test for `ownerName` substitution.

**Comment by finestructure** [comment]
The tests exposed a few little edge cases but it's all sorted now.

Thanks again @at-syot , this has been really helpful!

**Comment by at-syot** [comment]
> The tests exposed a few little edge cases but it's all sorted now.
> 
> Thanks again @at-syot , this has been really helpful!

Oh, the thx should go to you guys, i didn't do much haha. the true is i'm not sure about an expected result and how much i can make an edit to the function, i mean, am i allow to do this ? that's why i open draft pull request first to update, something like that.
So thank you again to lets me do this issue and the improved code is look good to.

**Comment by at-syot** [comment]
Just to confirm, after review the changed code, i assume that Package's collection name will stay the same with owner(short name), Only overview change to use owner_name's (multiple owner's full-name) instead. Am i understand correct ?

**Comment by finestructure** [comment]
No, the name is also changing. You can see it in the validation of the new test that I added, `test_generate_ownerName`:

```swift
          XCTAssertEqual(res.name, "Packages by Foo Org")
         XCTAssertEqual(res.overview, "A collection of packages authored by Foo Org from the Swift Package Index")
```

What I did though was ensure that you can still provide your own name and overview to `generate` and they'll be used as is. Only if the parameters are unset (`nil`) will these changes take effect.

Does that make sense?

**Comment by at-syot** [comment]
> No, the name is also changing. You can see it in the validation of the new test that I added, `test_generate_ownerName`:
> 
> ```swift
>          XCTAssertEqual(res.name, "Packages by Foo Org")
>          XCTAssertEqual(res.overview, "A collection of packages authored by Foo Org from the Swift Package Index")
> ```
> 
> What I did though was ensure that you can still provide your own name and overview to `generate` and they'll be used as is. Only if the parameters are unset (`nil`) will these changes take effect.
> 
> Does that make sense?

Ah, i see it now.

**Comment by finestructure** [comment]
I should add that I ended up making what perhaps looks like a lot of changes on top of your changes but it's really just re-arranging a few things and making the tests pass.

The tests are a great guide and will often instruct you what to still need to do to complete a task.

For instance here a lot of the additional changes were due to the second filter option `Filter.urls([String])`. Fixing the related tests made me realise that we need to re-arrange this a bit more, but the basic flow remained unchanged.

**Review by daveverwer — APPROVED** [review]
Fantastic work @at-syot! This is a great step forward.
